packages/@react-aria/interactions/src/usePress.ts

```diff
@@ -602,15 +602,54 @@ function createEvent(target: HTMLElement, e: EventBase): EventBase {
   };
 }
 
+interface Point {
+  x: number,
+  y: number
+}
+
+interface PointRect {
+  top: number,
+  right: number,
+  bottom: number,
+  left: number
+}
+
 interface EventPoint {
   clientX: number,
-  clientY: number
+  clientY: number,
+  width?: number,
+  height?: number,
+  radiusX?: number,
+  radiusY?: number
+}
+
+function getPointClientRect(point: EventPoint): PointRect {
+  let offsetX = (point.width / 2) || point.radiusX || 0;
+  let offsetY = (point.height / 2) || point.radiusY || 0;
```

Comment:
Do we need to check if `point.width`/`point.height` are `null` here? Feels weird to rely on NaN coercion behavior...

Reply:
Hah, I was actually quite surprised that TS has allowed this. I completely agree that relying on NaN coercion feels weird but at the same i've chosen this as other versions included more nesting and such and looked more complicated in the end than it really is.

Maybe I should just add a comment above those lines that this relies on NaN coercion?